yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -0,0 +1,316 @@
+	}
+	return args
+}
+
+func parseFleetConfigFile(filePath string) ([]AssessMigrationDBConfig, error) {
+	log.Infof("parsing fleet config file %q", filePath)
+	var dbConfigs []AssessMigrationDBConfig
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 {
+			continue
+		}
+		dbConfig := parseFleetConfigLine(line)
+		dbConfigs = append(dbConfigs, dbConfig)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return dbConfigs, nil
+}
+
+/*
+Format: covers both SSL and non-SSL cases
+
+	<dbtype>,<hostname>,<port>,<service_name>,<sid>,<tns_alias>,<username>,<password>,<schema>
+*/
+func parseFleetConfigLine(line string) AssessMigrationDBConfig {
+	config := strings.Split(line, ",")
+	return AssessMigrationDBConfig{
+		DbType:      config[0],
+		Host:        config[1],
+		Port:        config[2],
+		ServiceName: config[3],
+		SID:         config[4],
+		TnsAlias:    config[5],
+		User:        config[6],
+		Password:    config[7],
+		Schema:      config[8],
+	}
+}
+
+func getAssessmentExportDirPath(dbConfig AssessMigrationDBConfig) string {
+	switch {
+	case dbConfig.SID != "":
+		return fmt.Sprintf("%s/%s-%s-export-dir", bulkAssessmentDir, dbConfig.SID, dbConfig.Schema)
+	case dbConfig.ServiceName != "":
+		return fmt.Sprintf("%s/%s-%s-export-dir", bulkAssessmentDir, dbConfig.ServiceName, dbConfig.Schema)
+	case dbConfig.TnsAlias != "":
+		return fmt.Sprintf("%s/%s-%s-export-dir", bulkAssessmentDir, dbConfig.TnsAlias, dbConfig.Schema)
+	default:
+		panic(fmt.Sprintf("unable to get export dir for assessment of schema-%s", dbConfig.Schema))
+	}
+}
+
+func getDatabaseIdentifier(dbConfig AssessMigrationDBConfig) string {
+	switch {
+	case dbConfig.SID != "":
+		return dbConfig.SID
+	case dbConfig.ServiceName != "":
+		return dbConfig.ServiceName
+	case dbConfig.TnsAlias != "":
+		return dbConfig.TnsAlias
+	default:
+		return ""
+	}
+}
+
+//go:embed templates/bulkAssessmentReport.template
+var bulkAssessmentHtmlTmpl string
+
+func generateBulkAssessmentReport(dbConfigs []AssessMigrationDBConfig) error {
+	log.Infof("generating bulk assessment report")
+	for _, dbConfig := range dbConfigs {
+		exportDirPath := getAssessmentExportDirPath(dbConfig)
+		assessmentReportPath := filepath.Join(exportDirPath, "assessment", "reports", "assessmentReport.html")
+		assessmentReportRelPath, err := filepath.Rel(bulkAssessmentDir, assessmentReportPath)
+		if err != nil {
+			return fmt.Errorf("failed to get relative path for %s schema assessment report: %w", dbConfig.Schema, err)
+		}
+		var assessmentDetail = AssessmentDetail{
+			Schema:             dbConfig.Schema,
+			DatabaseIdentifier: getDatabaseIdentifier(dbConfig),
+			ReportPath:         assessmentReportRelPath,
+			Status:             COMPLETE,
+		}
+		if !utils.FileOrFolderExists(assessmentReportPath) {
```

Comment:
What if the command fails after generating the report? It should still be considered an ERROR, I think. We should probably get this information (or error or not) from the command.Run output, rather than interpreting it based on the presence of assessment report.

Reply:
hmm we can do that, will have to store all the results/err for each assessment.

Currently, i am doing two checks - assessmentReportExistsOrNot + checking MSR.MigrationAssessmentDone flag, which i thought is also full proof since the flag is set at the end of assess-migration command.
What do you think?

Reply:
using metadb flag should work! But here, you seem to only be checking if assessmentReportExistsOrNot?

Reply:
yeah got confused this with the same check at other place, there i am doing both.

Will add that here also.

Reply:
added to TO, will do it in the pending PR.

Reply:
done in the other PR - https://github.com/yugabyte/yb-voyager/pull/1624